docs/api/public.rst

```diff
@@ -77,6 +77,15 @@ Data
     GNUPlotFormat
     DiskIO
 
+DataSet
+~~~~~~~
+
+.. autosummary::
+   :toctree: generated/
+
```

Comment:
it is worth it to include DataSet here as well, at least for the time being. Also, we should definitely add all the `load_by_id`, `plot_by_id`, `initialise_database`, etc. stuff. Are they in already or not?

Reply:
This is really only baby steps. The whole API documentation needs a significant reorg. I was planning to do that gradually. In multiple prs